### DISCUSSION TITLE
Using JGit with browser JVM hits CORS restriction

### DISCUSSION THREAD

**Discussion by reportmill** [eclipse-jgit/jgit#85] [Q&A]
I have a unique use case - I'm trying to use JGit in the browser (using CheerpJ JVM) to clone a GitHub repository, and it fails to do a repository clone due to CORS restriction:

```
Access to fetch at 'https://github.com/reportmill/SnapDemos.git/info/refs?service=git-upload-pack'
from origin 'http://localhost:8080' has been blocked by CORS policy: Response to preflight request
doesn't pass access control check: No 'Access-Control-Allow-Origin' header is present on the
requested resource.
```

I suppose I really need GitHub to add the allow-origin response header, which I think they do in the GitHub API. Or I need to somehow make JGit use GitHub API instead of Git-Protocol. But otherwise I don't think I can use JGit for clone, push or fetch. 

I'm using JGit 5.13 because I'm temporarily limited to Java 8, but I don't think that will matter. Any ideas?

(My use case is a free IDE for the browser, [SnapCode](https://reportmill.com/SnapCode) ).

**Comment by msohn** [comment]
Searching for this question I found some interesting posts

- discussion on the gitlab tracker how to enable clone from browser: https://gitlab.com/gitlab-org/gitlab/-/issues/20590
- CORS proxy to proxy clone and push requests: https://github.com/isomorphic-git/cors-proxy, also see https://github.com/isomorphic-git/isomorphic-git

**Reply by reportmill** [reply]
Thanks for this - those links are helpful to flush out my understanding of CORS. I have actually been trying a corsproxy, but it has the same issues for some reason (I think it has problems with the CORS 'preflight' fetch).

I will probably end up having to do some kind of proxy server solution - CheerpJ has some options, I was just hoping there was a way around it without me having to pay for server time, since SnapCode is free.
